Formula/crc.rb

```diff
@@ -0,0 +1,28 @@
+class Crc < Formula
+  desc "OpenShift 4 cluster on your local machine"
+  homepage "https://code-ready.github.io/crc/"
+  url "https://github.com/code-ready/crc.git",
+      :tag      => "1.0.0",
+      :revision => "575079bff638214dc46d831caaef1c08927346ec"
+  head "https://github.com/code-ready/crc.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GO111MODULE"] = "on"
+
+    # Determine install OS and architecture
+    os = `go env GOOS`.strip
+    arch = `go env GOARCH`.strip
+    os = "macos" if os == "darwin"
+
+    # Make and install binary
+    system "make", "out/#{os}-#{arch}/crc"
+    bin.install "out/#{os}-#{arch}/crc"
+  end
+
+  test do
+    # Simple test, as most actions require sudo as they interact with hyperkit
+    assert_match /^crc version: #{version}/, shell_output("#{bin}/crc version")
```

Comment:
It might be good to have some functional test to assert the formula's correctness.

Reply:
As noted, this is only a basic test as the functional operations require sudo.

Reply:
We could add another test, that fails due to lack of permission…

Reply:
@fxcoudert Do you have an example of another formula that does this? Every other formula that I can find that requires sudo (e.g. bettercap, tcptraceroute, vsftpd, etc) all use a basic `version` or `--help` operation as a test.

Reply:
`boringtun`, `glooctl`, and others.

From the standard “what is a good test” text I have stored:

---

We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This is very helpful! Thank you @fxcoudert. I've added a better test.